pkg/net/net.go

```diff
@@ -0,0 +1,16 @@
+package net
+
+import (
+	"net"
+	"time"
+)
+
+// WaitForPort wait for successful network connection
+func WaitForPort(proto string, ip string, port string, timeout time.Duration) {
+	for {
+		_, err := net.DialTimeout(proto, ip+":"+port, timeout)
+		if err == nil {
```

Comment:
won't the error be non-nil if we hit the timeout, therefore making the timeout useless?